GameFiles/_integration.gsc

```diff
@@ -38,13 +34,35 @@ init()
 
     level.clientCommandCallbacks = [];
     level.clientCommandRusAsTarget = [];
+    level.overrideMethods = [];
+
+    wait 0.1;
+    InitializeGameMethods();
+
+    [[level.overrideMethods["SetDvarIfUninitialized"]]]( level.eventBus.inVar, "" );
+    [[level.overrideMethods["SetDvarIfUninitialized"]]]( level.eventBus.outVar, "" );
+    [[level.overrideMethods["SetDvarIfUninitialized"]]]( "sv_iw4madmin_integration_enabled", 1 );
+    [[level.overrideMethods["SetDvarIfUninitialized"]]]( "sv_iw4madmin_integration_debug", 0 );
+
+    level.logger = spawnstruct();
+    InitializeLogger();
+
+    //The gamename dvar seems to be an IW only thing..
+    if (level.eventBus.gamename == "") {
+        version = getDvar( "version" );
+        switch( version )
+	    {
+            case "Call of Duty Multiplayer - Ship COD_T5_S MP build 7.0.189 CL(1022875) CODPCAB-V64 CEG Wed Nov 02 18:02:23 2011 win-x86":
+                level.eventBus.gamename = "T5MP";
+            break;
+        }
```

Comment:
instead of doing switch here let's create a dictionary with the version pointing to the game name as this could in theory could get quite large
eg:
```
level.gameNames = [];
level.gameNames["Call of Duty Multiplayer - Ship COD_T5_S MP build 7.0.189 CL(1022875) CODPCAB-V64 CEG Wed Nov 02 18:02:23 2011 win-x86"] = "T5MP";
```

The default will be an empty string if the dict key doesn't exist so we can catch any logic there